app/packs/src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariationsUtils.js

```diff
@@ -0,0 +1,126 @@
+import cloneDeep from 'lodash/cloneDeep';
+
+function getMaterialData(material, requestedUnit) {
+  let value = null;
+  let unit = 'None';
+  if (requestedUnit === 'Equiv') {
+    value = material.equivalent ?? null;
+    unit = 'Equiv';
+  } else if (requestedUnit === 'Amount') {
+    value = material.amount_g ?? null;
+    unit = 'g';
+  }
+  const aux = {
+    coefficient: material.coefficient ?? null,
+    isReference: material.reference ?? false,
+    referenceAmount: material.reference ?? false ? material.amount_g ?? null : null,
```

Comment:
Kann man das hier vielleicht als `material.reference ?? (material.amount_g ?? null)` (kA ob man klammern muss) schreiben? Fallbacks und Ternary zusammen liest sich echt nicht gut

Reply:
Und was die ganzen Fallbacks betrifft: du fällst ja immer auf null zurück, das heißt das einzige wovor dich diese ?? schützen ist dass die Properties undefined sind. Kann das überhaupt auftreten wenn die API dafür sorgt, dass immer mindestens der Key da is?

Reply:
> Kann man das hier vielleicht als `material.reference ?? (material.amount_g ?? null)` (kA ob man klammern muss) schreiben? Fallbacks und Ternary zusammen liest sich echt nicht gut

Wenn `material.reference` `true` oder `false` ist (einen dieser Werte erwarte ich), gibt `material.reference ?? (material.amount_g ?? null)` nicht `material.amount_g` zurueck (sondern `true` bzw. `false`), daher der ternary.
Stimme dir allerdings vollstaendig zu, liesst sich bescheiden^^

Reply:
> Und was die ganzen Fallbacks betrifft: du fällst ja immer auf null zurück, das heißt das einzige wovor dich diese ?? schützen ist dass die Properties undefined sind. Kann das überhaupt auftreten wenn die API dafür sorgt, dass immer mindestens der Key da is?

`material` kommt aus dem JS Reaction Modell (https://github.com/ComPlat/chemotion_ELN/blob/7fe6e5834a484cbb8c70995fdbcf32d492864700/app/packs/src/models/Reaction.js).
Ich habe waehrend der Implementierung einige Ueberschungen mit undefinierten keys auf `material` erlebt, daher die defensive Vorgehensweise mit den `null` fallbacks.